torchtune/utils/collate.py

```diff
@@ -71,3 +71,57 @@ def padded_collate(
             value=padding_idx,
         )
     return input_ids, labels
+
+
+def padded_collate_dpo(
+    batch: List[TokenPair],
+    padding_idx: int = 0,
+    ignore_idx: int = -100,
+) -> Dict[str, torch.Tensor]:
+    """Pad a batch of sequences for Direct Preference Optimization (DPO).
+
+    This function takes a batch of sequences, where each sequence is represented
+    as a dictionary with multiple key-value pairs. Each key corresponds to a different
+    sequence component, such as input_ids or labels. The function pads each sequence
+    component independently to the longest sequence length in the batch.
+
+    Args:
+        batch (List[TokenPair]): A list of dictionaries, where each dictionary
+            represents a sequence with multiple components including 'chosen_input_ids',
```

Comment:
nit: I would make the language stronger here saying that these fields are required in the batch